### ISSUE TITLE
New folder changing name corrupts stack

### ISSUE THREAD

**Issue by lundman** [lundman/ZFSin#3]
Creating a new folder, and leaving the name `New Folder` works, and renaming it to another name as a separate process also works. But creating a new folder, and changing the name as part of the same task will corrupt stack.

```
if (vnode_isdir(ZTOV(szp))) {
		if ((error = zfs_rename_lock(szp, tdzp, sdzp, &zl)))
```
sets up the `zl` structure, which is from `kmem_alloc()`. Inside is mostly `NULLs`.

Then the return of this call
```
	error = dmu_tx_assign(tx, waited ? TXG_WAITED : TXG_NOWAIT);
```

corrupts the inside of `zl`. Cause unknown.

**Comment by lundman** [comment]
temporary work around e77cf594d216e4c2b90b26d674d2a9f229d83d04

**Comment by lundman** [comment]
revert e77cf59

corrected with 6dc4bc1cb93adf6717676293cb8de6dbc56cefdf
